Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/trim.h

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2021 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Andreas Fabri
+//
+#ifndef CGAL_POLYGON_MESH_PROCESSING_TRIM_H
+#define CGAL_POLYGON_MESH_PROCESSING_TRIM_H
+
+#include <CGAL/license/Polygon_mesh_processing/corefinement.h>
+
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/Polygon_mesh_processing/clip.h>
+#include <CGAL/Polygon_mesh_processing/locate.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Polygon_mesh_processing/orient_polygon_soup.h>
+#include <CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h>
+#include <CGAL/Polygon_mesh_processing/connected_components.h>
+
+#include <vector>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+/**
+  * \ingroup PMP_corefinement_grp
+  *
+  * trims `tm` at a polyline that must be close to the surface mesh.
+  * @tparam TriangleMesh a model of `MutableFaceGraph`, `HalfedgeListGraph`, and `FaceListGraph`.
+  *                      An internal property map for `CGAL::vertex_point_t` must be available.
+  * @tparam PointRange a range of 3D points
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param tm input triangulated surface mesh that will also contain one output mesh
+  * @param other the second output mesh
+  * @param pr the point range for a closed polyline  that is close to `tm`. The first and last point must be equal.
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+**/
+
+template <class TriangleMesh,
+          class PointRange,
+          class NamedParameters = parameters::Default_named_parameters>
+bool trim(TriangleMesh& tm,
+          TriangleMesh& other,
+          const PointRange& pr,
+          const NamedParameters& np = parameters::default_values())
+{
+
+  typedef typename GetGeomTraits<TriangleMesh, NamedParameters>::type     Geom_traits;
+  typedef typename Geom_traits::Point_3                                   Point_3;
+  typedef typename Geom_traits::Vector_3                                  Vector_3;
+  typedef typename boost::graph_traits<TriangleMesh>::face_descriptor     face_descriptor;
+
+  typedef CGAL::AABB_face_graph_triangle_primitive<TriangleMesh>          AABB_face_graph_primitive;
+  typedef CGAL::AABB_traits<Geom_traits, AABB_face_graph_primitive>       AABB_face_graph_traits;
+  typedef CGAL::AABB_tree<AABB_face_graph_traits>                         Tree;
+  typedef Face_location<TriangleMesh, double>                        Face_location;
+
+  Tree tree;
+  build_AABB_tree(tm, tree);
+
+  int n = std::distance(pr.begin(), pr.end());
+  const double extrusion_distance = 0.1;
```

Comment:
this looks arbitrary to me

Reply:
What you can do is keep the normal direction per vertex and increase the extrusion while the trimer mesh does not intersect entirely the mesh. You can check that for each "quad" corresponding to a segment of the polyline